flake.nix

```diff
@@ -368,6 +375,7 @@
 
               '') partitions)}
 
+            args+=(${ builtins.concatStringsSep " " ( pkgs.lib.attrsets.mapAttrsToList (addr: src: "\"${src}@${addr}\"") extraBinaryBlobs ) })
```

Comment:
Maybe this would be a good use for either `lib.strings.escapeShellArg` (https://github.com/NixOS/nixpkgs/blob/master/lib/strings.nix#L443) or `lib.strings.toShellVar` (https://github.com/NixOS/nixpkgs/blob/master/lib/strings.nix#L485)